src/services/inlayHints.ts

```diff
@@ -421,6 +444,258 @@ export function provideInlayHints(context: InlayHintsContext): InlayHint[] {
         });
     }
 
+    function typeToInlayHintParts(type: Type): InlayHintDisplayPart[] | string {
+        if (!shouldUseInteractiveInlayHints(preferences)) {
+            return printTypeInSingleLine(type);
+        }
+
+        const flags = NodeBuilderFlags.IgnoreErrors | TypeFormatFlags.AllowUniqueESSymbolType | TypeFormatFlags.UseAliasDefinedOutsideCurrentScope;
+        const typeNode = checker.typeToTypeNode(type, /*enclosingDeclaration*/ undefined, flags);
+        Debug.assertIsDefined(typeNode, "should always get typenode");
+
+        const parts: InlayHintDisplayPart[] = [];
+        visitDisplayPart(typeNode);
+        function visitDisplayPart(node: Node) {
+            if (!node) {
+                return;
+            }
+
+            if (preferences.includeInlayFunctionParameterTypeHints && isFunctionLikeDeclaration(node) && hasContextSensitiveParameters(node)) {
+                visitFunctionLikeForParameterType(node);
+            }
+            if (preferences.includeInlayFunctionLikeReturnTypeHints && isSignatureSupportingReturnAnnotation(node)) {
+                visitFunctionDeclarationLikeForReturnType(node);
+            }
+
+            switch (node.kind) {
+                case SyntaxKind.AnyKeyword:
+                case SyntaxKind.BigIntKeyword:
+                case SyntaxKind.BooleanKeyword:
+                case SyntaxKind.IntrinsicKeyword:
+                case SyntaxKind.NeverKeyword:
+                case SyntaxKind.NumberKeyword:
+                case SyntaxKind.ObjectKeyword:
+                case SyntaxKind.StringKeyword:
+                case SyntaxKind.SymbolKeyword:
+                case SyntaxKind.UndefinedKeyword:
+                case SyntaxKind.UnknownKeyword:
+                case SyntaxKind.VoidKeyword:
+                case SyntaxKind.ThisType:
+                    parts.push({ text: tokenToString(node.kind)! });

```

Comment:
Instead of this very-likely-to-not-be-updated list, why not just move this out of the switch and write the following?

```ts
const tokenString = tokenToString(node.kind);
if (tokenString !== undefined) {
     parts.push({ text: tokenString });
     return;
}
```

Reply:
Genius.